File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem42_DeleteNodeInABSTTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem33_MaximumDepthOfBinaryTree.TreeNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem42_DeleteNodeInABSTTest {

    private final Problem42_DeleteNodeInABST test = new Problem42_DeleteNodeInABST();

    @Test
    void deleteNode() {
        TreeNode root = new TreeNode(5);
        root.left = new TreeNode(3);
        root.left.left = new TreeNode(2);
        root.left.right = new TreeNode(4);
        root.right = new TreeNode(6);
        root.right.right = new TreeNode(7);

        assertEquals(4, test.deleteNode(root, 3).left.val);
    }
}

File: JavaScript/leetcode75/src/test/Problem42_DeleteNodeInABST.test.js
const deleteNode = require("../main/Problem42_DeleteNodeInABST");

//  Definition for a binary tree node.
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

test("should delete the specified node", () => {
  let root = new TreeNode(5);
  root.left = new TreeNode(3);
  root.left.left = new TreeNode(2);
  root.left.right = new TreeNode(4);
  root.right = new TreeNode(6);
  root.right.right = new TreeNode(7);

  expect(deleteNode(root, 3).left.val).toBe(4);
});
